pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1348,6 +1349,7 @@ func (ml *MutableListener) build(builder *ListenerBuilder, opts buildListenerOpt
 			if len(opt.httpOpts.statPrefix) == 0 {
 				opt.httpOpts.statPrefix = strings.ToLower(ml.Listener.TrafficDirection.String()) + "_" + ml.Listener.Name
 			}
+			opt.httpOpts.port = *opts.port
```

Comment:
This only works for outbound listener, you need to set port for in bound in function `buildSidecarInboundHTTPOpts`

Reply:
For the inbound case, please look at https://github.com/istio/istio/pull/40862/files#diff-4afa570feba990275e28fa8ca0b5be122e9ff5d6f0beee9d2e8f54cb57d46613R650